mesonbuild/modules/fs.py

```diff
@@ -69,6 +70,23 @@ def expanduser(self, state: 'ModuleState', args: T.Sequence[str], kwargs: dict)
             raise MesonException('fs.expanduser takes exactly one argument.')
         return ModuleReturnValue(str(Path(args[0]).expanduser()), [])
 
+    @stringArgs
+    @noKwargs
+    @FeatureNew('fs.expand', '0.57.0')
+    def expand(self, state: 'ModuleState', args: T.Sequence[str], kwargs: dict) -> ModuleReturnValue:
+        base = self._resolve_dir(state)
+        result = set()    # type: T.Set[str]
+        for p in args:
+            arg = Path(p).expanduser()
+            # Path.glob does not support absolute paths
+            if arg.is_absolute():
+                drive = Path(arg.parts[0])
+                p = str(arg.relative_to(drive))
+                result.update(str(x) for x in drive.glob(p))
+            else:
+                result.update(str(x.relative_to(base)) for x in base.glob(p))
+        return ModuleReturnValue(sorted(result), [])
+
```

Comment:
I really feel like we should have an `mlog.warn(once=true)` here about the dangers of globbing.